openedx/openedx-platform
# LX-24 Extract static paths

Adds an optional `static_paths` argument to `static_replace.replace_static_urls` which allows the caller to receive the collected static paths and their replacements.

This change allows us to parse a given XBlock's OLX to locate related course assets, so they can be included when the XBlock is transferred to [Blockstore](https://openedx.atlassian.net/wiki/spaces/AC/pages/737149430/Blockstore+Design).

**JIRA tickets**: OC-5133, LX-24, OSPR

**Sandbox URL**:

* LMS: https://pr18855.sandbox.opencraft.hosting/
* Studio: https://studio-pr18855.sandbox.opencraft.hosting/

**Merge deadline**: ASAP

**Testing instructions**:

I believe this change is sufficiently tested by the unit tests.

The linked sandbox can be used to ensure that there are no adverse effects on static files served by the LMS/Studio.

**Author notes and concerns**:

1. This isn't an ideal implementation, since it relies on side-effects to achieve its goal.  But I wanted to keep the change minimal, so as not to harm static asset generation.

**Reviewers**
- [ ] @ormsbee
